scala/build/Dockerfile.latest

```diff
@@ -0,0 +1,124 @@
+    make build-essential
+# TODO Build by installing package with specific version
+# RUN apt-get update && apt-get install -y --no-install-recommends \  
+#     git=<version> curl=<version> iputils-ping=<version> telnet=<version> vim=<version> \  
+#     gnupg=<version> unzip=<version> wget=<version> python3=<version> python3-venv=<version> \  
+#     postgresql-client=<version> default-jre=<version> default-jdk=<version> \  
+#     apt-transport-https=<version> ca-certificates=<version> \  
+#     software-properties-common=<version> chromium-browser=<version> xvfb=<version> \  
+#     python3-full=<version> make=<version> build-essential=<version> \  
+#     && rm -rf /var/lib/apt/lists/* 
+
+# Create a virtual environment
+RUN python3 -m venv /venv
+
+# Activate the virtual environment
+ENV PATH="/venv/bin:$PATH"
+
+# Upgrade pip within the virtual environment
+RUN pip install --upgrade pip
+
+# Copy the requirements file into the container
+COPY scala/build/requirements.txt .
+
+# Install dependencies from the requirements file
+RUN pip install -r requirements.txt
+
+
+# Install Yarn
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && \
+    apt-get install -y yarn
+
+# Install Node.js and utilities
+RUN apt-get install -y nodejs npm && \
+    npm install -g n
+
+# RUN npm install -g npm@latest
+RUN n stable
+
+# Install AWS CLI
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
+    unzip awscliv2.zip && \
+    ./aws/install && \
+    rm -rf awscliv2.zip ./aws
+
+# Install Docker and Docker Compose
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - && \
+    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" && \
+    apt-get update && \
+    apt-get install -y docker-ce docker-ce-cli containerd.io && \
+    curl -L "https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose && \
+    chmod +x /usr/local/bin/docker-compose
+
+# TODO build docker engine with specific version
+# https://docs.docker.com/engine/install/ubuntu/
+# RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - && \
+#     add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" && \
+#     apt-get update && \
+#     apt-get install -y --no-install-recommends \
+#     docker-ce=${DOCKER_VERSION} \
+#     docker-ce-cli=${DOCKER_VERSION} \
+#     containerd.io && \
+#     curl -L "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose && \
+#     chmod +x /usr/local/bin/docker-compose && \
+#     rm -rf /var/lib/apt/lists/*
```

Comment:
@coderabbitai build failure with latest version
 > [11/16] RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - &&     add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" &&     apt-get update &&     apt-get install -y --no-install-recommends     docker-ce=27.3.1     docker-ce-cli=27.3.1     containerd.io &&     curl -L "https://github.com/docker/compose/releases/download/2.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose &&     chmod +x /usr/local/bin/docker-compose &&     rm -rf /var/lib/apt/lists/*:
3.439   docker-buildx-plugin
3.439 
3.439 Package docker-ce is not available, but is referred to by another package.
3.439 This may mean that the package is missing, has been obsoleted, or
3.439 is only available from another source
3.439 However the following packages replace it:
3.439   docker-ce-cli
3.439 
3.440 E: Version '27.3.1' for 'docker-ce' was not found
3.440 E: Version '27.3.1' for 'docker-ce-cli' was not found

Reply:
@coderabbitai ubuntu:24.10